simconvoi.cc

```diff
@@ -5497,6 +5504,40 @@ void convoi_t::calc_sum_friction_weight() {
 	}
 }
 
+// jump to other line's schedule
+void convoi_t::change_line_to_next_if_needed() 
+{
+	linehandle_t l = get_schedule()->get_next_line();
+	// check and update next_line
+	if(  l.is_bound() && l->get_schedule()->get_count()>1 && l->get_schedule()->at(0).pos!=get_schedule()->at(get_schedule()->get_count()-1).pos ) {
+		if(  get_line().is_bound()  ) {
+			get_line()->get_schedule()->set_next_line(l);
+		}
+		get_schedule()->set_next_line(l);
+	}
```

Comment:
TODO: make it clear that these block is for unsetting the next line. Isn't the check by `check_next_line_valid` enough?